### ISSUE TITLE
CycloneDX ingestion, check for invalid bom-refs

### ISSUE THREAD

**Issue by ctron** [trustification/trustify#551]
Currently we simply accept the `bom-ref` references as they are. However, it may be, that there are broken refs or duplicates, like with SPDX.

We should provide a user friendly error in such a case, and allow any import process to continue, still raising the issue in a report or response. Following what we already do for SPDX.

**Comment by jcrossley3** [comment]
Still relevant in light of recent 1.6 upgrade?

**Comment by ctron** [comment]
Yes it is. Right now this still triggers an internal error IIRC.

**Comment by ctron** [comment]
I changed the priority to P2. Simply because this will not create faulty data, but "only" provide the user with a bad error message.

**Comment by JimFuller-RedHat** [comment]
might be useful to have a test demonstrating this issue

**Comment by ctron** [comment]
I'll create a test and maybe add a fix for it too.
